File: Java/Solutions/Problem015/Stream.java
package Problem015;

import java.util.List;
import java.util.Random;

/**
 * Functions for a stream of T
 */
public class Stream<T> {
    public static <T> T GetRandomElementFromStream(List<T> stream) {
        var random = new Random();
        var randomElement = random.nextInt(stream.size());

        return stream.get(randomElement);
    }
}

File: C#/Solutions/Problem015/Stream.cs
using System;
using System.Collections.Generic;

namespace DailyCodingProblems.Solutions.Problem015
{
    /// <summary>
    /// Functions for a stream of <see cref="T"/>
    /// </summary>
    /// <typeparam name="T"></typeparam>
    public class Stream<T>
    {
        public static T GetRandomElementFromStream(IReadOnlyList<T> stream)
        {
            var random = new Random();
            var randomElement = random.Next(0, stream.Count);

            return stream[randomElement];
        }
    }
}
